### PR TITLE
Stabilize the pipeline services API

### PR DESCRIPTION
**Pull request by bourdakos1**
- Only export a migrate and validate function
- fix huge performance issues
- fix potential infinite loop
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/PipelineController/index.ts [hunk 1]
```diff
@@ -260,64 +184,184 @@ class PipelineController extends CanvasController {
...
+            stroke-dasharray: 13;
+            `,
+        },
+      },
+      true
+    );
+  }
+
+  resetStyles() {
+    // @ts-ignore
```
Comment (marthacryan) [inline packages/pipeline-editor/src/PipelineController/index.ts:223]: Why is this necessary?

**Review by marthacryan — APPROVED** [review]
LGTM

File: packages/pipeline-editor/src/PipelineController/index.ts (hunk 1 above)
Comment (bourdakos1) [reply packages/pipeline-editor/src/PipelineController/index.ts:223]: Oh right, I forgot I need to update the types

Comment (bourdakos1) [reply packages/pipeline-editor/src/PipelineController/index.ts]: fixed

**Review by lresende — APPROVED** [review]
